grails-app/services/org/pih/warehouse/order/OrderService.groovy

```diff
@@ -869,7 +869,13 @@ class OrderService {
                     if (currentLocation.isAccountingRequired() && !code) {
                         throw new IllegalArgumentException("Budget code is required.")
                     }
-                    BudgetCode budgetCode = BudgetCode.findByCode(code)
+
+                    // There can be more than one budget code with the same code, despite the fact that the code is unique from the domain perspective.
+                    // As a fix for that we would like to have only one active budget code with the same code, so we have to find only active
+                    // budget codes. In case when there is only one budget code, and this one is inactive we have to throw a validation error,
```

Comment:
if you say so, why don't you throw this exception if the `foundBudgetCodes.first()` would be inactive?

Reply:
It's threw a few lines below